superset/sso_auth.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from datetime import timezone, datetime
+import jwt
+from cryptography.hazmat.backends import default_backend
+from cryptography.x509 import load_pem_x509_certificate
+
+from flask import g, redirect, request
+from flask_login import login_user
+from superset import security_manager
+from . import config
+
+def get_publickey():
+    key = config.KNOX_PUBLIC_KEY
+    #coverting to pem cert string
+    PEM_HEADER = "-----BEGIN CERTIFICATE-----\n"
+    PEM_FOOTER = "\n-----END CERTIFICATE-----"
+    cert_str = PEM_HEADER + key + PEM_FOOTER
+    cert_obj = load_pem_x509_certificate(cert_str.encode('ascii'), default_backend())
+    return cert_obj.public_key()
+
+def get_token_contents(token):
+    contents = jwt.decode(token,get_publickey())
+    return contents
+
+def _find_user(username, sm):
+    """extracted from flask_appbuilder.security.manager.BaseSecurityManager.find_user"""
+    user = sm.find_user(username)
+    if not user:
+        user = security_manager.add_user(
+                username= username,
+                first_name= username,
+                last_name=username,
+                email=username + '@email.notfound',
+                role=security_manager.find_role("Admin")
+            )
+    return user
+
+def parse_hadoop_jwt():
+    if(config.KNOX_SSO_ENABLED):
+        logging.info("Attaching JWT handler")
+        jwt_token = request.cookies.get("hadoop-jwt", None)
```

Comment:
This name of token can be configurable. Can we take this from config and default to `hadoop-jwt` if not provided